src/main/java/com/cairn/waypoint/dashboard/endpoints/HomeworkQuestionLinkedProtocolTemplate/LinkHomeworkQuestionAndProtocolTemplateEndpoint.java

```diff
@@ -0,0 +1,94 @@
+package com.cairn.waypoint.dashboard.endpoints.HomeworkQuestionLinkedProtocolTemplate;
+import com.cairn.waypoint.dashboard.entity.HomeworkQuestion;
+import com.cairn.waypoint.dashboard.entity.HomeworkQuestionLinkedProtocolTemplate;
+import com.cairn.waypoint.dashboard.entity.ProtocolTemplate;
+import com.cairn.waypoint.dashboard.repository.HomeworkQuestionRepository;
+import com.cairn.waypoint.dashboard.repository.HomeworkQuestionLinkedProtocolTemplatesRepository;
+import com.cairn.waypoint.dashboard.repository.ProtocolTemplateRepository;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import java.util.Optional;
+
+@Slf4j
+@RestController
+@RequestMapping("/api/protocol-template-homework")
```

Comment:
Also, this request path should look more like this: `POST api/protocol-template/homework-question`. We are adding a homework question to a protocol template